Actions/AL-Go-Helper.ps1

```diff
@@ -2360,4 +2360,81 @@ function Get-PackageVersion($PackageName) {
     else {
         throw "Package $PackageName is not in the list of packages"
     }
-}
\ No newline at end of file
+}
+
+function InstallAzModuleIfNeeded {
+    Param(
+        [string] $moduleName
+    )
+
+    if (Get-Module -Name $moduleName) {
+        # Already installed
+        return
+    }
+    if ($isWindows) {
+        # GitHub hosted Windows Runners have AZ PowerShell module saved in C:\Modules\az_*
+        # Remove AzureRm modules from PSModulePath and add AZ modules
+        if (Test-Path 'C:\Modules\az_*') {
+            $azModulesPath = Get-ChildItem 'C:\Modules\az_*' | Where-Object { $_.PSIsContainer }
+            if ($azModulesPath) {
+              Write-Host "Adding AZ module path: $($azModulesPath.FullName)"
+              $ENV:PSModulePath = "$($azModulesPath.FullName);$(("$ENV:PSModulePath".Split(';') | Where-Object { $_ -notlike 'C:\\Modules\Azure*' }) -join ';')"
+            }
+        }
+    }
+    else {
+        # Linux runners have AZ PowerShell module saved in /usr/share/powershell/Modules/Az.*
+    }
+    $azModule = Get-Module -name $moduleName -ListAvailable | Select-Object -First 1
+    if ($azModule) {
+        Write-Host "$moduleName Module is available in version $($azModule.Version)"
+        Write-Host "Using $moduleName version $($azModule.Version)"
+    }
+    else {
+        $AzModule = Get-InstalledModule -Name $moduleName -ErrorAction SilentlyContinue
+        if ($AzModule) {
+            Write-Host "$moduleName version $($AzModule.Version) is installed"
+        }
+        else {
+            Write-Host "Installing and importing $moduleName"
+            Install-Module $moduleName -Force
+        }
+    }
+    Import-Module $moduleName -DisableNameChecking -WarningAction SilentlyContinue | Out-Null
+}
+
+function ConnectAz {
```

Comment:
It would work when authenticating towards azure resources (on Azure KeyVault) - but we would still need to do it ourselves when publishing apps to appsource or to Business Central environments, as these are not azure resources - and we would have to have a condition on the job that the secret exists and the keyvaultname setting exists.
I don't think it is worth it.

Reply:
>  but we would still need to do it ourselves when publishing apps to appsource or to Business Central environments, as these are not azure resources 

Why is that? 🤔 It's still Entra authentication I assume

Reply:
When you authenticate (with federated credentials, clientsecrets or other means) you get an accesstoken for a specific audience. When you use the Azure/Login - the audience is azure resource management.
That accesstoken is no good for Business Central API's, Admin Center APIs, AppSource Validation APIs etc. - it only means that az CLI is now authenticated.